File: languages/java/src/main/java/com/bloock/sdk/entity/identity/StringAttribute.java
package com.bloock.sdk.entity.identity;

import com.bloock.sdk.bridge.proto.BloockIdentityEntities;

/**
 * Represents an attribute with a string value.
 */
public class StringAttribute extends Attribute<String> {
  /**
   * Creates a new StringAttribute instance with the provided key and value.
   * 
   * @param id
   * @param value
   */
  public StringAttribute(String id, String value) {
    super(id, value);
  }

  public static StringAttribute fromProto(BloockIdentityEntities.StringAttribute res) {
    return new StringAttribute(res.getId(), res.getValue());
  }

  public BloockIdentityEntities.StringAttribute toProto() {
    return BloockIdentityEntities.StringAttribute.newBuilder()
        .setId(this.id)
        .setValue(this.value)
        .build();
  }
}

File: languages/go/entity/identityV2/string_attribute.go
package identityV2

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

type StringAttribute struct {
	Attribute
}

func NewStringAttribute(key string, value string) StringAttribute {
	a := Attribute{
		Id:    key,
		Value: value,
	}
	return StringAttribute{
		Attribute: a,
	}
}

func NewStringAttributeFromProto(s *proto.StringAttribute) StringAttribute {
	if s == nil {
		return StringAttribute{}
	}
	a := Attribute{
		Id:    s.GetId(),
		Value: s.GetValue(),
	}
	return StringAttribute{
		Attribute: a,
	}
}

func (s StringAttribute) ToProto() *proto.StringAttributeV2 {
	return &proto.StringAttributeV2{
		Id:    s.Attribute.Id,
		Value: s.Attribute.Value.(string),
	}
}
